### PR TITLE
All: print vj.trip.id

### PR DESCRIPTION
**Pull request by pbougue**
JIRA: http://jira.canaltp.fr/browse/NAVP-319

### CODE REVIEW

File: source/navitia-proto [hunk 1]
```diff
@@ -1 +1 @@
-Subproject commit 485a92b929d8e6d164592a2f4cadf2fb23c6bb82
+Subproject commit 83c3be775c450322874c509e816ebe4035f70632
```
Comment (pbougue) [inline source/navitia-proto]: waiting CanalTP/navitia-proto/pull/16 merge for exact SHA-1

File: source/ed/docker_tests/ed_integration_tests.cpp [hunk 2]
```diff
@@ -122,4 +124,8 @@ BOOST_FIXTURE_TEST_CASE(fusio_test, ArgsFixture) {
     BOOST_CHECK(navitia::contains(headsigns.get_vj_from_headsign("vehiclejourney3"), vj_vec[5]));
 
     BOOST_CHECK_EQUAL(data.pt_data->meta_vjs.size(), 3);
+    for(auto& mvj : data.pt_data->meta_vjs) {
+        BOOST_CHECK_EQUAL(data.pt_data->meta_vjs[mvj->uri], mvj.get());
+        BOOST_CHECK_EQUAL(data.pt_data->meta_vjs[navitia::Idx<nt::MetaVehicleJourney>(mvj->idx)], mvj.get());
```
Comment (TeXitoi) [inline source/ed/docker_tests/ed_integration_tests.cpp]: maybe you can typedef that MvjIdx somewhere as it is done in raptor_utils.h for the Idx.

File: source/ed/docker_tests/ed_integration_tests.cpp [hunk 3]
```diff
@@ -122,4 +124,8 @@ BOOST_FIXTURE_TEST_CASE(fusio_test, ArgsFixture) {
     BOOST_CHECK(navitia::contains(headsigns.get_vj_from_headsign("vehiclejourney3"), vj_vec[5]));
 
     BOOST_CHECK_EQUAL(data.pt_data->meta_vjs.size(), 3);
+    for(auto& mvj : data.pt_data->meta_vjs) {
```
Comment (TeXitoi) [inline source/ed/docker_tests/ed_integration_tests.cpp]: spacing

File: source/type/pb_converter.cpp [hunk 4]
```diff
@@ -639,6 +639,15 @@ void fill_pb_object(const nt::ValidityPattern* vp, const nt::Data&,
     validity_pattern->set_days(vp->days.to_string());
 }
 
+void fill_pb_object(const nt::MetaVehicleJourney* nav_mvj,
```
Comment (TeXitoi) [inline source/type/pb_converter.cpp:642]: fill_pb_object takes a const nt::Data& as second argument

File: source/type/type.h [hunk 5]
```diff
@@ -990,7 +990,7 @@ struct MetaVehicleJourney: public Header {
     std::map<std::string, AssociatedCalendar*> associated_calendars;
 
     template<class Archive> void serialize(Archive & ar, const unsigned int ) {
-        ar & base_vj & adapted_vj & real_time_vj & associated_calendars;
+        ar & idx & uri & base_vj & adapted_vj & real_time_vj & associated_calendars;
```
Comment (TeXitoi) [inline source/type/type.h:993]: Oups ;-)

File: source/jormungandr/tests/ptref_tests.py [hunk 6]
```diff
@@ -579,3 +579,12 @@ def test_headsign_display_info_route_schedules(self):
         display_info = route_schedules[0]['table']['headers'][0]['display_informations']
         eq_(display_info['headsign'], "vjA")
         assert {"A00", "vjA"} == set(display_info['headsigns'])
+
+    def test_trip_id_vj(self):
```
Comment (antoine-de) [inline source/jormungandr/tests/ptref_tests.py:583]: you need to change the `is_valid_vehicle_journey` method

File: source/ed/docker_tests/ed_integration_tests.cpp (hunk 2 above)
Comment (xlqian) [inline source/ed/docker_tests/ed_integration_tests.cpp]: Is `navitia::Idx<nt::MetaVehicleJourney>` a casting?

Comment (antoine-de) [inline source/ed/docker_tests/ed_integration_tests.cpp]: nop it is a constructor.

BTW you shouldn't provide the idx, you should use to constructor with a MVJ to be typesafe

File: source/type/pb_converter.cpp (hunk 4 above)
Comment (pbougue) [inline source/type/pb_converter.cpp:642]: Usually yes, but we don't see the need for it in a close future so I vote to let it that way.

File: source/ed/docker_tests/ed_integration_tests.cpp (hunk 2 above)
Comment (pbougue) [inline source/ed/docker_tests/ed_integration_tests.cpp]: @antoine-de will do that

File: source/jormungandr/tests/ptref_tests.py (hunk 6 above)
Comment (pbougue) [inline source/jormungandr/tests/ptref_tests.py:583]: done

**Pull request merged**